analyses/molecular-subtyping-MB/02-compare-classes.Rmd

```diff
@@ -0,0 +1,221 @@
+                               scrollX = TRUE,
+                               dom = 'tpi',
+                               initComplete = JS("function(settings, json) {",
+                                                 "$(this.api().table().header()).css({'background-color':
+                                            '#004467', 'color': '#fff'});","}"))
+  )
+}
+```
+
+#### Details:
+
+1. % Accuracy is currently being calculated by matching observed and expected subtypes where expected subtype info is available. In case of ambiguous subtypes, we treat it as a match if the observed subtype matches with any one of the expected subtypes.
+
+2. Consensus tabs: Molecular subtype labels are assigned only when the two classifiers agree, leaving all other samples as unclassified. 
+
+## {.tabset .tabset-fade}
+
+### MM2S (batch-corrected)
+
+```{r, echo = TRUE, warning = FALSE, message = FALSE}
+# merge observed and expected subtypes
+mm2s.corrected <- obs.class[[1]]
+mm2s.corrected <- exp.class %>%
+  left_join(mm2s.corrected, by = c('Kids_First_Biospecimen_ID' = 'sample')) %>%
+  mutate(match = str_detect(pathology_subtype, best.fit))
+
+# calculate % accuracy
+mm2s.corrected.acc <- mm2s.corrected %>%
+  filter(!is.na(pathology_subtype)) %>%
+  group_by(match) %>%
+  summarise(n = n()) %>%
+  mutate(Accuracy = paste0(n/sum(n)*100, '%')) %>%
+  filter(match) %>%
+  .$Accuracy
+print(paste0("Accuracy: ", mm2s.corrected.acc))
+
+# output table
+viewDataTable(mm2s.corrected)
+```
+
+### MM2S (uncorrected)
+
+```{r, echo = TRUE, warning = FALSE, message = FALSE}
+# merge observed and expected subtypes
+mm2s.uncorrected <- obs.class[[2]]
+mm2s.uncorrected <- exp.class %>%
+  inner_join(mm2s.uncorrected, by = c('Kids_First_Biospecimen_ID' = 'sample')) %>%
+  mutate(match = str_detect(pathology_subtype, best.fit))
+
+# % accuracy
+mm2s.uncorrected.acc <- mm2s.uncorrected %>%
+  filter(!is.na(pathology_subtype)) %>%
+  group_by(match) %>%
+  summarise(n = n()) %>%
+  mutate(Accuracy = paste0(n/sum(n)*100, '%')) %>%
+  filter(match) %>%
+  .$Accuracy
+print(paste0("Accuracy: ", mm2s.uncorrected.acc))
+
+# output table
+viewDataTable(mm2s.uncorrected)
+```
+
+### medulloPackage (batch-corrected)
+
+```{r, echo = TRUE, warning = FALSE, message = FALSE}
+# merge observed and expected subtypes
+medullo.classifier.corrected <- obs.class[[3]]
+medullo.classifier.corrected <- exp.class %>%
+  inner_join(medullo.classifier.corrected, by = c('Kids_First_Biospecimen_ID' = 'sample')) %>%
+  mutate(match = str_detect(pathology_subtype, best.fit))
+
+# % accuracy
+medullo.classifier.corrected.acc <- medullo.classifier.corrected %>%
+  filter(!is.na(pathology_subtype)) %>%
+  group_by(match) %>%
+  summarise(n = n()) %>%
+  mutate(Accuracy = paste0(n/sum(n)*100, '%')) %>%
+  filter(match) %>%
+  .$Accuracy
+print(paste0("Accuracy: ", medullo.classifier.corrected.acc))
+
+# output table
+viewDataTable(medullo.classifier.corrected)
```

Comment:
A lot of this code is repeated between different classifier-dataset combinations. That suggests to me that it may be useful to wrap up joining with the expected subtype information and accuracy calculation in a custom function.

Reply:
I have added individual functions here: 
https://github.com/komalsrathi/OpenPBTA-analysis/blob/mb-class-nb/analyses/molecular-subtyping-MB/02-compare-classes.Rmd#L18